lib/libspl/include/sys/simd.h

```diff
@@ -491,6 +491,24 @@ zfs_altivec_available(void)
 #endif
 	return (has_altivec);
 }
+static inline boolean_t
+zfs_vsx_available(void)
+{
+	boolean_t has_vsx = B_FALSE;
+#if defined(__ALTIVEC__) && !defined(__FreeBSD__)
+	sighandler_t savesig;
+	savesig = signal(SIGILL, sigillhandler);
+	if (setjmp(env)) {
+		signal(SIGILL, savesig);
+		has_vsx = B_FALSE;
+	} else {
+		__asm__ __volatile__("xxinsert 0,0,0\n");
```

Comment:
I believe I contributed this originally, but fascinatingly, it doesn't seem to compile for me, but twiddling it to `xxinsertw` behaves correctly for me...and I can't find much evidence this should have worked, so I have questions for past-me, and how this compiled before (or on the Debian 8 PPC builder...)

...the PPC builder seems to be reporting:
```
WARNING: ".zfs_blake3_compress_xof_sse2" [/var/lib/buildbot/slaves/zfs2/Debian_8_ppc64__BUILD_/build/zfs/module/zfs.ko] undefined!
WARNING: ".zfs_blake3_compress_in_place_sse41" [/var/lib/buildbot/slaves/zfs2/Debian_8_ppc64__BUILD_/build/zfs/module/zfs.ko] undefined!
WARNING: ".zfs_blake3_hash_many_sse41" [/var/lib/buildbot/slaves/zfs2/Debian_8_ppc64__BUILD_/build/zfs/module/zfs.ko] undefined!
WARNING: ".zfs_blake3_compress_in_place_sse2" [/var/lib/buildbot/slaves/zfs2/Debian_8_ppc64__BUILD_/build/zfs/module/zfs.ko] undefined!
WARNING: ".zfs_blake3_hash_many_sse2" [/var/lib/buildbot/slaves/zfs2/Debian_8_ppc64__BUILD_/build/zfs/module/zfs.ko] undefined!
WARNING: ".zfs_blake3_compress_xof_sse41" [/var/lib/buildbot/slaves/zfs2/Debian_8_ppc64__BUILD_/build/zfs/module/zfs.ko] undefined!
```
which I'm surprised isn't a build failure, since I thought the kernel passed -Werror and we passed -Wall on those...

But on the plus side:
```
Test: /home/rich/zfs_blake3_oncemore/tests/zfs-tests/tests/functional/checksum/setup (run as root) [00:02] [PASS]
Test: /home/rich/zfs_blake3_oncemore/tests/zfs-tests/tests/functional/checksum/run_edonr_test (run as root) [00:01] [PASS]
Test: /home/rich/zfs_blake3_oncemore/tests/zfs-tests/tests/functional/checksum/run_sha2_test (run as root) [00:09] [PASS]
Test: /home/rich/zfs_blake3_oncemore/tests/zfs-tests/tests/functional/checksum/run_skein_test (run as root) [00:07] [PASS]
Test: /home/rich/zfs_blake3_oncemore/tests/zfs-tests/tests/functional/checksum/run_blake3_test (run as root) [00:05] [PASS]
Test: /home/rich/zfs_blake3_oncemore/tests/zfs-tests/tests/functional/checksum/filetest_001_pos (run as root) [01:32] [PASS]
Test: /home/rich/zfs_blake3_oncemore/tests/zfs-tests/tests/functional/checksum/filetest_002_pos (run as root) [00:43] [PASS]
Test: /home/rich/zfs_blake3_oncemore/tests/zfs-tests/tests/functional/checksum/cleanup (run as root) [00:02] [PASS]
$ uname -m
ppc64le
$ uname -r
5.15.41
```

(The kernel is patched for `mmu_feature_keys`, but otherwise vanilla.)

Gonna keep poking it on AArch64 and ppc64le just to be safe, since there's no test coverage in the CI...